packages/core/src/balance/Balance.utils.ts

```diff
@@ -33,8 +66,97 @@ export function convertToTxUnit(balance: BN, power: number): BN {
+    }
+    return input
+  }
+  if (
+    typeof input === 'number' ||
+    (typeof input === 'bigint' && input.toString) ||
+    (typeof input === 'object' && input instanceof BN && input.toString)
+  ) {
+    return input.toString()
+  }
+  throw new Error('could not convert to String')
 }
 
-export const TRANSACTION_FEE = convertToTxUnit(new BN(125), -9)
+/**
+ * Converts the given [[BalanceNumber]] to the femto KILT equivalent.
+ *
+ * @param input [[BalanceNumber]] to convert.
+ * @param unit Metric prefix of the given [[BalanceNumber]].
+ * @returns Exact BN representation in femtoKilt, to use in transactions and calculations.
+ * @throws Unknown metricPrefix, or if the input has too many decimal places for it's unit.
+ */
+export function toFemtoKilt(
+  input: BalanceNumber,
+  unit: MetricPrefix = 'KILT'
+): BN {
+  const stringRepresentation = balanceNumberToString(input)
+
+  if (!Prefixes.has(unit)) {
+    throw new Error('Unknown metric prefix')
+  }
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const unitVal = Prefixes.get(unit)!
+  const negative = stringRepresentation.substring(0, 1) === '-'
+
+  const [integer, fraction] = negative
+    ? stringRepresentation.substring(1).split('.')
+    : stringRepresentation.split('.')
+  if (fraction && fraction.length > unitVal + 15) {
+    throw new Error(
+      `Too many decimal places: input with unit ${unit} and value ${stringRepresentation} exceeds the ${
+        unitVal + 15
+      } possible decimal places by ${fraction.length - unitVal + 15}`
+    )
+  }
+  const fractionBN = fraction
+    ? convertToTxUnit(new BN(fraction), unitVal - fraction.length)
+    : new BN(0)
+  const resultingBN = convertToTxUnit(new BN(integer), unitVal).add(fractionBN)
+
+  return resultingBN.mul(new BN(negative ? -1 : 1))
+}
+
+/**
+ * Converts the given [[BalanceNumber]] to a human readable number with metric prefix and Unit.
+ * This function uses the polkadot formatBalance function,
+ * it's output can therefore be formatted via the polkadot formatting options.
+ *
+ * @param input [[BalanceNumber]] to convert from Femto Kilt.
+ * @param decimals [[number]] Set the minimum decimal places in the formatted localized output, default is 4.
+ * @param options [[BalanceOptions]] for internationalization and formatting.
+ * @returns String representation of the given [[BalanceNumber]] with unit und metric prefix.
+ */
+export function fromFemtoKilt(
+  input: BalanceNumber,
+  decimals = 4,
+  options?: BalanceOptions
+): string {
+  const inputBN = new BN(balanceNumberToString(input))
+  const formatted = formatKiltBalance(inputBN, options)
+  const [number, ...rest] = formatted.split(' ')
+  const negative = number.startsWith('-')
+  const localeNumber = new Intl.NumberFormat(options?.locale, {
+    minimumFractionDigits: 4,
+  }).format(Number(negative ? number.slice(1) : number))
+  return `${negative ? '-' : ''}${localeNumber.slice(
```

Comment:
`slice` can take a negative number as second parameter, than you would not have to use `localeNumber.length`

Reply:
Not really possible here, as in the case of decimals equaling 4, calling slice(0,0) returns empty string ''

I could of course concat a space to be able to offset the second parameter to max -1